validation.go

```diff
@@ -442,12 +442,16 @@ func (v *subSchema) validateCommon(currentSubSchema *subSchema, value interface{
 
 	// format:
 	if currentSubSchema.format != "" {
-		if !FormatCheckers.IsFormat(currentSubSchema.format, value) {
+		if err := FormatCheckers.IsFormat(currentSubSchema.format, value); err != nil {
+			details := ErrorDetails{"format": currentSubSchema.format}
+			if err != ErrBadFormat {
+				details["error"] = err
```

Comment:
NIT but can we potentially change `error` to `fmtError`? Right now, `ErrorDetails` is overloaded with a lot of other key names like `expected`, `reference`, `format`, etc. so `error` becomes extremely unclear in what context this key should be used or overloaded.

Or maybe call it `customMsg` or `customError` if we want this custom error injection in other validation checks in the future.

Reply:
ErrorDetails holds "a map of details specific to each error". In this case it's a `DoesNotMatchFormatError` type error.

I think the `error` key fits in very well with the existing general key names like `expected`, `reference`, `format`, etc. @johandorland would be better able to judge though.

In looking into this I noticed that I hadn't updated the relevant comment in errors.go, so I've done that now in 4a6c4b6.